apps/extension/src/state/persist.ts

```diff
@@ -70,6 +74,39 @@ function syncLocal(changes: Record<string, chrome.storage.StorageChange>, set: S
       }),
     );
   }
+
+  if (changes['grpcEndpoint']) {
+    const stored = changes['grpcEndpoint'].newValue as
+      | StorageItem<LocalStorageState['grpcEndpoint']>
+      | undefined;
+    set(
+      produce((state: AllSlices) => {
+        state.network.grpcEndpoint = stored?.value ?? state.network.grpcEndpoint;
+      }),
+    );
+  }
+
+  if (changes['knownSites']) {
+    const stored = changes['knownSites'].newValue as
+      | StorageItem<LocalStorageState['knownSites']>
+      | undefined;
+    set(
+      produce((state: AllSlices) => {
+        state.connectedSites.knownSites = stored?.value ?? state.connectedSites.knownSites;
+      }),
+    );
+  }
+
+  if (changes['frontendUrl']) {
+    const stored = changes['frontendUrl'].newValue as
+      | StorageItem<LocalStorageState['frontendUrl']>
+      | undefined;
+    set(
+      produce((state: AllSlices) => {
+        state.connectedSites.frontendUrl = stored?.value ?? state.connectedSites.frontendUrl;
+      }),
+    );
+  }
```

Comment:
Adding a few missing properties to what gets synced on storage changes.

We really need to refactor this persist implementation to auto-sync between state and local storage. Otherwise, we have sort of spaghetti code. For example, [`setFrontendUrl`](https://github.com/penumbra-zone/web/pull/915/files#diff-7ec170797ae75c84fd4d1e8acd3cdfbb74fe61bd02d734c2199446b68b34ef74R26-R28) only sets a key in local storage; it doesn't actually update the Zustand state, because this persist implementation syncs from local storage to state. But that's hard for devs to understand.

IMO, ideally, our persistence implementation should have a mapping of keys in local storage to their location in state, and automatically handle the syncing whenever store values are updated — rather than store methods having to call `local.set()`. But that refactor seemed out of scope for this PR.